homeassistant/components/recorder/models.py

```diff
@@ -102,11 +102,15 @@ class States(Base):  # type: ignore
     entity_id = Column(String(255))
     state = Column(String(255))
     attributes = Column(Text)
-    event_id = Column(Integer, ForeignKey("events.event_id"), index=True)
+    event_id = Column(
+        Integer, ForeignKey("events.event_id", ondelete="SET NULL"), index=True
```

Comment:
```suggestion
        Integer, ForeignKey("events.event_id", ondelete="CASCADE"), index=True
```

I think we can `CASCADE` here since we want to delete both.

Reply:
I agree. Did you test this already?

Reply:
Yes